Kerberos.NET/Client/Krb5CredentialCache.cs

```diff
@@ -0,0 +1,600 @@
+         * 
+         * Versions 3 and 4 always use big-endian byte order.
+         * 
+         * After the two-byte version indicator, the file has three parts:
+         *  * the header (in version 4 only),
+         *  * the default principal name,
+         *  * and a sequence of credentials
+         * 
+         * header ::=
+         *     It begins with a 16-bit integer giving the length of the entire header, followed by a sequence of fields. 
+         *     Each field consists of a 16-bit tag, a 16-bit length, and a value of the given length.
+         * 
+         * principal ::=
+         *     name type (32 bits) [omitted in version 1]
+         *     count of components (32 bits) [includes realm in version 1]
+         *     realm (data)
+         *     component1 (data)
+         *     component2 (data)
+         *     ...
+         * 
+         * data ::=
+         *     length (32 bits)
+         *     value (length bytes)
+         *     
+         * credential ::=
+         *     client (principal)
+         *     server (principal)
+         *     keyblock (keyblock)
+         *     authtime (32 bits)
+         *     starttime (32 bits)
+         *     endtime (32 bits)
+         *     renew_till (32 bits)
+         *     is_skey (1 byte, 0 or 1)
+         *     ticket_flags (32 bits)
+         *     addresses (addresses)
+         *     authdata (authdata)
+         *     ticket (data)
+         *     second_ticket (data)
+         * 
+         * keyblock ::=
+         *     enctype (16 bits) [repeated twice in version 3]
+         *     data
+         * 
+         * addresses ::=
+         *     count (32 bits)
+         *     address1
+         *     address2
+         *     ...
+         * 
+         * address ::=
+         *     addrtype (16 bits)
+         *     data
+         * 
+         * authdata ::=
+         *     count (32 bits)
+         *     authdata1
+         *     authdata2
+         *     ...
+         * 
+         * authdata ::=
+         *     ad_type (16 bits)
+         *     data
+         */
+
+        public int Version { get; } = ExpectedVersion;
+
+        public IDictionary<Krb5CredentialCacheTag, ReadOnlyMemory<byte>> Header { get; } = new Dictionary<Krb5CredentialCacheTag, ReadOnlyMemory<byte>>();
+
+        public PrincipalName DefaultPrincipalName { get; set; }
+
+        public ICollection<Krb5Credential> Credentials { get; } = new List<Krb5Credential>();
+
+        public bool FastAvailable { get; set; }
+
+        public string PreAuthConfiguration { get; set; }
+
+        public PaDataType PreAuthType { get; set; }
+
+        public string ProxyImpersonator { get; set; }
+
+        internal void Read(byte[] cache)
+        {
+            this.Credentials.Clear();
+            this.Header.Clear();
+
+            var buffer = new NdrBuffer(cache, align: false);
+
+            var magic = buffer.Read(1)[0];
+
+            if (Magic != magic)
```

Comment:
```suggestion
            if (magic != Magic)
```
Don't use [yoda style](https://de.wikipedia.org/wiki/Yoda_Conditions) here 😉